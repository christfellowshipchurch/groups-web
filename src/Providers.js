import React from 'react';
import { ThemeProvider } from 'styled-components';
import PropTypes from 'prop-types';

import { defaultTheme } from './ui-kit/theme';

const Providers = ({ children }) => (
  <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
);

Providers.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default Providers;
