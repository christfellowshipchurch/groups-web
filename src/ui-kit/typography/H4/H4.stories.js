import React from 'react';
import { storiesOf } from '@storybook/react';

import H4 from '.';

storiesOf('ui-kit/typography/H4', module)
  .add('default', () => (
    <H4>
      {
        '"God is most glorified in us when we are most satisfied in Him" ― John Piper'
      }
    </H4>
  ))
  .add('isLoading', () => (
    <H4 isLoading showLoadingAnimation>
      {
        '"God is most glorified in us when we are most satisfied in Him" ― John Piper'
      }
    </H4>
  ))
  .add('withMargins', () => <H4 withMargins>Padded Heading Big and Long</H4>)
  .add('withMargins + isLoading', () => (
    <H4 isLoading withMargins showLoadingAnimation>
      {
        '"God is most glorified in us when we are most satisfied in Him" ― John Piper'
      }
    </H4>
  ));
