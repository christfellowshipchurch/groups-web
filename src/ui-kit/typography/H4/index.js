import React from 'react';
import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';

import { Line } from '../../Placeholder';

const verticalRhythm = css`
  margin: ${({ padded, theme }) =>
    padded
      ? `${theme.helpers.verticalRhythm(0.66)} 0 ${theme.helpers.verticalRhythm(
          0.5
        )} 0`
      : 'initial'};
`;

const StyledH4 = styled.h4.attrs({
  className: 'ui-kit-typography-H4-StyledH4',
})`
  font-size: 1rem;
  font-weight: 700;
  line-height: ${({ theme }) => theme.helpers.verticalRhythm(1)};

  ${verticalRhythm}
`;

const Placeholder = styled(Line).attrs({
  className: 'ui-kit-H1-Placeholder',
})`
  height: 1rem; /* Should match H4 font-size */

  ${verticalRhythm}
`;

const H4 = ({ children, isLoading, padded, ...props }) => (
  <Placeholder isLoading={isLoading} padded={padded} showLoadingAnimation>
    <StyledH4 padded={padded} {...props}>
      {children}
    </StyledH4>
  </Placeholder>
);

H4.defaultProps = {
  isLoading: false,
};

H4.propTypes = {
  children: PropTypes.node.isRequired,
  isLoading: PropTypes.bool,
  padded: PropTypes.bool,
};

H4.displayName = 'ui-kit.typography.H4';

export default H4;
