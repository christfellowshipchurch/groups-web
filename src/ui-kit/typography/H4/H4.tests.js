import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Providers from '../../../Providers';

import H4 from '.';

describe('the H4 component', () => {
  it('should render', () => {
    const tree = renderer.create(
      <Providers>
        <H4>Default H4 text</H4>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should accept and render passed in styles', () => {
    const salmonText = { color: 'salmon' };
    const tree = renderer.create(
      <Providers>
        <H4 style={salmonText}>Salmon text</H4>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render with vertical rhythm (padded)', () => {
    const tree = renderer.create(
      <Providers>
        <H4 padded>H4 text with vertical rhythm styled via margin.</H4>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render a loading state', () => {
    const tree = renderer.create(
      <Providers>
        <H4 isLoading>Default H4 text</H4>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render a loading state with vertical rhythm (padded)', () => {
    const tree = renderer.create(
      <Providers>
        <H4 isLoading padded>
          Default H4 text
        </H4>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
