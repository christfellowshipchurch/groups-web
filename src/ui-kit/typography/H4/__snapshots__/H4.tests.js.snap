// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the H4 component should accept and render passed in styles 1`] = `
.c0 {
  font-size: 1rem;
  font-weight: 700;
  line-height: 1.5rem;
  margin: initial;
  color: salmon;
}

<h4
  className="c0 ui-kit-typography-H4-StyledH4"
>
  Salmon text
</h4>
`;

exports[`the H4 component should render 1`] = `
.c0 {
  font-size: 1rem;
  font-weight: 700;
  line-height: 1.5rem;
  margin: initial;
}

<h4
  className="c0 ui-kit-typography-H4-StyledH4"
>
  Default H4 text
</h4>
`;

exports[`the H4 component should render a loading state 1`] = `
.c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: calc(1.5rem - 1rem) 0;
}

<mark
  className="c0 show-loading-animation c1 ui-kit-H4-Placeholder ui-kit-Placeholder-Line custom-placeholder"
/>
`;

exports[`the H4 component should render a loading state via context 1`] = `
.c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: calc(1.5rem - 1rem) 0;
}

<mark
  className="c0 show-loading-animation c1 ui-kit-H4-Placeholder ui-kit-Placeholder-Line custom-placeholder"
/>
`;

exports[`the H4 component should render a loading state with vertical rhythm (withMargins) 1`] = `
.c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: 0.99rem 0 0.75rem 0;
}

<mark
  className="c0 show-loading-animation c1 ui-kit-H4-Placeholder ui-kit-Placeholder-Line custom-placeholder"
/>
`;

exports[`the H4 component should render with vertical rhythm (withMargins) 1`] = `
.c0 {
  font-size: 1rem;
  font-weight: 700;
  line-height: 1.5rem;
  margin: 0.99rem 0 0.75rem 0;
}

<h4
  className="c0 ui-kit-typography-H4-StyledH4"
>
  H4 text with vertical rhythm styled via margin.
</h4>
`;
